examples/google-cloud-speech/output/google-cloud-speech/lib/google/cloud/speech/v1/speech_client.rb

```diff
@@ -0,0 +1,425 @@
+          #   retries, etc.
+          # @return [Google::Gax::Operation]
+          # @raise [Google::Gax::GaxError] if the RPC is aborted.
+          # @example
+          #   require "google/cloud/speech"
+          #
+          #   speech_client = Google::Cloud::Speech.new version: :v1
+          #
+          #   op = speech_client.get_operation "-"
+          #
+          #   # Process error operations.
+          #   log_error op.error if op.error?
+          #
+          #   if op.done?
+          #     # Process completed operations.
+          #     log_finished op.response, op.metadata
+          #   else
+          #     # Process pending operations.
+          #     log_pending op.name, op.metadata
+          #   end
+          #
+          def get_operation name, options: nil
+            proto_op = @operations_client.get_operation name, options: options
+
+            Google::Gax::Operation.new(
+              proto_op,
+              @operations_client,
+              nil,
+              nil,
+              call_options: options
+            )
+          end
+
+          protected
+
+          def create_stub credentials, scopes
+            if credentials.is_a?(String) || credentials.is_a?(Hash)
+              updater_proc = Google::Cloud::Speech::V1::Credentials.new(credentials).updater_proc
+            elsif credentials.is_a? GRPC::Core::Channel
+              channel = credentials
+            elsif credentials.is_a? GRPC::Core::ChannelCredentials
+              chan_creds = credentials
+            elsif credentials.is_a? Proc
+              updater_proc = credentials
+            elsif credentials.is_a? Google::Auth::Credentials
+              updater_proc = credentials.updater_proc
+            end
+
+            # Allow overriding the service path/port in subclasses.
+            service_path = self.class::SERVICE_ADDRESS
+            port = self.class::DEFAULT_SERVICE_PORT
+            interceptors = self.class::GRPC_INTERCEPTORS
+            Google::Gax::Grpc.create_stub(
+              service_path,
+              port,
+              chan_creds: chan_creds,
+              channel: channel,
+              updater_proc: updater_proc,
+              scopes: scopes,
+              interceptors: interceptors,
+              &Google::Cloud::Speech::V1::Speech::Stub.method(:new)
+            )
+          end
+
+          def default_settings client_config, timeout, metadata, lib_name, lib_version
+            package_version = Gem.loaded_specs["google-cloud-speech"].version.version
+
+            google_api_client = "gl-ruby/#{RUBY_VERSION}"
+            google_api_client << " #{lib_name}/#{lib_version}" if lib_name
+            google_api_client << " gapic/#{package_version} gax/#{Google::Gax::VERSION}"
+            google_api_client << " grpc/#{GRPC::VERSION}"
+            google_api_client.freeze
+
+            headers = { "x-goog-api-client": google_api_client }
```

Comment:
@blowmage As you mentioned offline, I believe this hash key symbol style is not supported by Ruby < 2.2. Is that a problem?

Reply:
I'm not sure what the policy will be for dropping support for older rubies in the generator. It is likely that a google-gas 2.0 gem won't support those rubies, so I wouldn't worry too much about that.

However, this _may_ a problem because the hash key value is now a symbol instead of a string. There may be unintended consequences from changing the hash key value type.

Reply:
Because I extracted a method in the same commit, the diff is hard to follow, but here's the old line 146:

`headers = { :"x-goog-api-client" => google_api_client }`

You can see that the key value was a string, only the hash syntax is changed.

Reply:
Gotcha. This seems fine then, as long as the Ruby runtime requirements are clear and agreed upon.